folktables/folktables.py

```diff
@@ -141,7 +141,42 @@ def df_to_pandas(self, df, categories=None, dummies=False):
 
         return variables, target, group
 
-    @property 
+    def generate_categories(self, definition_df):
+        """
+        Generates a categories dictionary using the provided definition dataframe. Does not create a category mapping
+        for variable requiring the 2010 Public use microdata area code (PUMA) as these need an additional definition
```

Comment:
Clarify in the doc string what `definition_df` is and what assumptions you make about this dataframe. See other methods for how to describe arguments in a doc string.